packages/components/src/components/as-histogram-widget/utils/draw.service.ts

```diff
@@ -293,9 +296,31 @@ function _delayFn(_d, i) {
   return i;
 }
 
-export function conditionalFormatter(value) {
-  if (value > 0 && value < 1) {
-    return decimalFormatter(value);
+function getFloatPrecision(value) {
+  const expValue = value.toString();
+  const expPos = expValue.indexOf('.');
+  return expPos > -1 ? expValue.slice(expPos + 1).length : 0;
```

Comment:
Why should we slice the string? We can just get the difference from the decimal position and the original string length, right?